lib/global-styles.php

```diff
@@ -137,11 +137,23 @@ function_exists( 'gutenberg_is_edit_site_page' ) &&
 	// Copied from get_block_editor_settings() at wordpress-develop/block-editor.php.
 	$settings['__experimentalFeatures'] = $consolidated->get_settings();
 	if ( isset( $settings['__experimentalFeatures']['color']['palette'] ) ) {
-		$settings['colors'] = $settings['__experimentalFeatures']['color']['palette'];
+		$colors_by_origin   = $settings['__experimentalFeatures']['color']['palette'];
+		$settings['colors'] = isset( $colors_by_origin['user'] ) ?
+			$colors_by_origin['user'] : (
+				isset( $colors_by_origin['theme'] ) ?
+					$colors_by_origin['theme'] :
+					$colors_by_origin['core']
+			);
 		unset( $settings['__experimentalFeatures']['color']['palette'] );
```

Comment:
It seems we want this data in the client, to be accessed by  `useSetting( 'color.palette.origin' )`, correct?

Reply:
Yes in the future we want it but for now, we can not add it. The way we are back-compatible with plugins changing the old color settings is by using `unset( $settings['__experimentalFeatures']['color']['palette'] );`. If we don't do this ` $settings['__experimentalFeatures']['color']['palette']` will take priority over the old settings and these plugins break. That was the reason we were already doing `unset( $settings['__experimentalFeatures']['color']['palette'] );` and I don't think we can stop doing that here. I'm not sure how we are going to address this in the future.

Reply:
Hi @nosolosw I ended up removing these unsets I tested with the plugin Block Editor Colors and a theme without theme.json and the plugin still works as expected.

Reply:
According to the direction https://github.com/WordPress/gutenberg/pull/32358#issuecomment-853282220:

> The legacy settings at core/block-editor.settings.colors still hold only the colors in use by the UI controrls (either core or theme colors).

In addition to no longer unset the `settings.__experimentalFeatures.colors` we still need to retrofit the values to the old ones. Otherwise, what happens is that `settings.colors` holds incorrect data. Steps to reproduce with colors (same for gradients and font sizes):

- use WordPress 5.7, TT1-blocks, and activate this branch as the plugin
- go and see the values in the `core/block-editor` store under `settings.colors`
- the expected result was that it held the theme values but instead it had the default core colors

Reply:
We are now retrofitting the values to the old ones.